### ISSUE TITLE
Question about the DMT

### ISSUE THREAD

**Issue by whynot3333** [bozhenhhu/DMVGAE#1]
Hi Author,
Thank you for sharing the paper codes. I have a question about the DMT module. In the paper, you wrote that the prior graph is the given graph. In the code, the corresponding part is as follows: 
` if use_dmt == 'True':
                loss_dmt = LossClass(graph.x, graph.x, latent, latent)
                loss += balance * loss_dmt`
whether graph.x refers to the original feature data of the dataset?
For example, in Cora, the original feature dimension is 1433. Then, based on these original features, a probabilistic graph is constructed, and the embeddings generated by the encoder are aligned with its manifold. My current understanding is that the original features may not effectively distinguish nodes, so I find it a bit difficult to understand .
I am relatively new to the field of graph deep learning, so my understanding might be incorrect. I would greatly appreciate it if you could clarify this for me.
Thank you in advance for your time and help!

**Comment by bozhenhhu** [comment]
Thank you for your question and interest in our work!

In the case of Cora (and many standard graph datasets), the prior graph is not constructed from the original node features. Instead, the graph structure is already given as part of the dataset. This includes predefined edges that represent relationships between nodes.

In the code snippet you mentioned, graph.x does refer to the original node features (1433-dimensional in Cora). The prior graph's structure (edges) is fixed and provided by the dataset itself. The DMT module aligns the learned embeddings (from the encoder) with the manifold structure implied by this predefined prior graph. This alignment helps preserve the intrinsic relational structure of the data during embedding learning.

For more, you can refer to this paper, [DMEAG](https://www.sciencedirect.com/science/article/abs/pii/S0925231222011882) ,   and [DLME](https://github.com/zangzelin/code_ECCV2022_DLME), @zangzelin

I hope this clarifies your confusion!

**Comment by whynot3333** [comment]
> Thank you for your question and interest in our work!
> 
> In the case of Cora (and many standard graph datasets), the prior graph is not constructed from the original node features. Instead, the graph structure is already given as part of the dataset. This includes predefined edges that represent relationships between nodes.
> 
> In the code snippet you mentioned, graph.x does refer to the original node features (1433-dimensional in Cora). The prior graph's structure (edges) is fixed and provided by the dataset itself. The DMT module aligns the learned embeddings (from the encoder) with the manifold structure implied by this predefined prior graph. This alignment helps preserve the intrinsic relational structure of the data during embedding learning.
> 
> For more, you can refer to this paper, [DMEAG](https://www.sciencedirect.com/science/article/abs/pii/S0925231222011882) , and [DLME](https://github.com/zangzelin/code_ECCV2022_DLME), [@zangzelin](https://github.com/zangzelin)
> 
> I hope this clarifies your confusion!

Thank you for your reply. However, I still don’t quite understand why the original features inherently contain the intrinsic manifold structure. For example, in Cora, applying the t-SNE algorithm to the original features （1433-dimensional）produces a very scattered plot. Does this mean that the original features truly imply a meaningful manifold representation? This is the point I’m most confused about.

**Comment by bozhenhhu** [comment]
The key point is that the intrinsic manifold structure we care about is not derived from the original features alone—it is primarily defined by the given graph structure (the edges), not the node features themselves. The core idea is the graph structure preserving.

**Comment by whynot3333** [comment]
> The key point is that the intrinsic manifold structure we care about is not derived from the original features alone—it is primarily defined by the given graph structure (the edges), not the node features themselves. The core idea is the graph structure preserving.

My main confusion comes from this part of the code, where the distances are computed directly from the features without involving the graph structure (the edges). Do you mean that the distances computed from the features implicitly contain the graph structure?
`disInput = self._DistanceSquared(input_data, input_data)
        distlatent = self._DistanceSquared(latent_data, latent_data)`
`    def _DistanceSquared(self, x, y):
        return torch.pow(torch.cdist(x, y, p=2),2)`

**Comment by bozhenhhu** [comment]
Yes. The distances are computed from the features based on the graph structure.
This full pairwise distance matrix is certainly filtered using the graph’s edge structure (adjacency matrix) in the complete implementation. This step ensures that only distances between nodes connected by edges (or within a local neighborhood defined by the graph) are actually used in the loss calculation.
